meta-store/src/meta_store.rs

```diff
@@ -0,0 +1,163 @@
+/*
+    Quickwit
+    Copyright (C) 2021 Quickwit Inc.
+
+    Quickwit is offered under the AGPL v3.0 and as commercial software.
+    For commercial licensing, contact us at hello@quickwit.io.
+
+    AGPL:
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+mod single_file;
+
+use std::fmt::{Debug, Display};
+use std::io;
+use std::ops::Range;
+
+use async_trait::async_trait;
+use thiserror::Error;
+
+pub type IndexId = String;
+pub type SplitId = String;
+
+pub enum State {
+    // the splits is almost ready. Some of its files may have been uploaded in the storage.
+    Staged,
+    // the splits is ready and published.
+    Published,
+    // the split is scheduled to be deleted.
+    ScheduledForDeleted,
+}
+
+pub struct SplitMetaData {
+    // Split uri. In spirit, this uri should be self sufficient
+    // to identify a split.
+    // In reality, some information may be implicitly configure
+    // in the store uri resolver, such as the Amazon S3 region.
+    split_uri: String,
+
+    // Number of records (or document) in the split
+    num_records: u64,
+
+    // Weight in bytes of the split
+    size_in_bytes: u64,
+
+    // if a time field is available, the min / max timestamp in the segment.
+    time_range: Option<Range<u64>>,
+
+    // Number of merge this segment has been subjected to during its lifetime.
+    generation: usize,
+}
+
+/// A split manifest carries all meta data about a split
+/// and its files.pub
+pub struct SplitManifest {
+    // the index it belongs to or is destined to belong to.
+    id: IndexId,
```

Comment:
@mosuka  @fulmicoton @guilload  we need to agree on this. `id` might not be what we want I think. we can rename it to `index` and move `split_uri` from `SplitMetaData` to `SplitManifest` as the `Id`.
The Idea is when holding a SplitManifest, I don't have to go down to get the id or uri in order to resolve the storage

Reply:
@evanxg852000 @guilload @fulmicoton 
Structures have been updated.
Referring to the JSON in the comments below, I have added a new IndexMetaData to include the index information.
https://github.com/quickwit-inc/quickwit/issues/2#issuecomment-839954684
Please give me feedback.